### ISSUE TITLE
sort-keys should not encourage breaking code (with experimentalObjectRestSpread).

### ISSUE THREAD

**Issue by scott-m-sarsfield** [eslint/eslint#10261]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8
* **npm Version:** --

**What parser (default, Babel-ESLint, etc.) are you using?**
default, presumably

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 6,
        "sourceType": "script",
        "ecmaFeatures": {
            "experimentalObjectRestSpread": true
        }
    },
    "rules": {
        "constructor-super": 2,
        "no-case-declarations": 2,
        "no-class-assign": 2,
        "no-compare-neg-zero": 2,
        "no-cond-assign": 2,
        "no-console": 2,
        "no-const-assign": 2,
        "no-constant-condition": 2,
        "no-control-regex": 2,
        "no-debugger": 2,
        "no-delete-var": 2,
        "no-dupe-args": 2,
        "no-dupe-class-members": 2,
        "no-dupe-keys": 2,
        "no-duplicate-case": 2,
        "no-empty-character-class": 2,
        "no-empty-pattern": 2,
        "no-empty": 2,
        "no-ex-assign": 2,
        "no-extra-boolean-cast": 2,
        "no-extra-semi": 2,
        "no-fallthrough": 2,
        "no-func-assign": 2,
        "no-global-assign": 2,
        "no-inner-declarations": 2,
        "no-invalid-regexp": 2,
        "no-irregular-whitespace": 2,
        "no-mixed-spaces-and-tabs": 2,
        "no-new-symbol": 2,
        "no-obj-calls": 2,
        "no-octal": 2,
        "no-redeclare": 2,
        "no-regex-spaces": 2,
        "no-self-assign": 2,
        "no-sparse-arrays": 2,
        "no-this-before-super": 2,
        "no-undef": 2,
        "no-unexpected-multiline": 2,
        "no-unreachable": 2,
        "no-unsafe-finally": 2,
        "no-unsafe-negation": 2,
        "no-unused-labels": 2,
        "no-unused-vars": 2,
        "no-useless-escape": 2,
        "require-yield": 2,
        "use-isnan": 2,
        "valid-typeof": 2,
        "sort-keys": 2
    },
    "env": {}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
var a = 1, b = 2, c = {a: 4, d: 3}, d = 4, e = 5;

function f(){}

f({
	d,
	e,
	...c,
	a,
	b
});
```

Ran it on https://eslint.org/demo/

**What did you expect to happen?**
No linter errors.

**What actually happened? Please include the actual, raw output from ESLint.**
Error on line 9. `Expected object keys to be in ascending order. 'a' should be before 'e'.`

**Comment by platinumazure** [comment]
Hi @scott-m-sarsfield, thanks for the issue.

Hmm... I think I agree, this seems like a bug. We should not report on objects that use object spread.

**Comment by katerberg** [comment]
I'm working on a PR for this
